sample_consensus/include/pcl/sample_consensus/impl/ransac.hpp

```diff
@@ -55,77 +58,148 @@ pcl::RandomSampleConsensus<PointT>::computeModel (int)
   }
 
   iterations_ = 0;
-  int n_best_inliers_count = -INT_MAX;
-  double k = 1.0;
+  std::size_t n_best_inliers_count = 0;
+  double k = INFINITY;
 
   std::vector<int> selection;
   Eigen::VectorXf model_coefficients;
 
-  double log_probability  = std::log (1.0 - probability_);
-  double one_over_indices = 1.0 / static_cast<double> (sac_model_->getIndices ()->size ());
+  const double log_probability  = std::log (1.0 - probability_);
+  const double one_over_indices = 1.0 / static_cast<double> (sac_model_->getIndices ()->size ());
 
-  int n_inliers_count = 0;
+  std::size_t n_inliers_count;
```

Comment:
There is never a read before a write: the return value of `countWithinDistance` is written to it before the value is read in the if-statement. I actually removed the initialization with 0, because it is unnecessary and could IMO create a misunderstanding.

Reply:
This is never needed outside the "loop", in which case we can move this inside, right next to `countWithinDistance`, and it becomes thread_local. Same is true for `selection` and `model_coefficients` while `model_coefficients_` and `model_` are shared variables (acknowledged later in comments in the code) but not explicitly stated in the pragma. I don't know OpenMP that well, so please make a uniform choice that fits the situation.

Reply:
I could move `n_inliers_count` inside the loop, but I wouldn't do the same for `selection` and `model_coefficients`, because then they would have to be constructed and destructed every iteration. I could move them between `pragma omp parallel` and the start of the loop, but I don't really see an advantage then?
Regarding `model_coefficients_` and `model_`: static class members are shared by default. They cannot be written explicitly in the pragma `shared`-clause.

Reply:
Sounds good Sorry for the delay, have a happy new year. Thanks for the info 😄